### ISSUE TITLE
paddingTop/Bottom isn't applied on Input on Android

### ISSUE THREAD

**Issue by vbriand** [GeekyAnts/NativeBase#3783]
**Describe the bug**
Props `paddingTop` and `paddingBottom` aren't taken into account by the `<Input />` component on Android.

**To Reproduce**
Steps to reproduce the behaviour:
1. Create an `<Input />` component
2. Give it a small height, let's say 9
3. Add `paddingTop` and `paddingBottom` props
4. The padding doesn't change from the default one

**Expected behaviour**
The padding works like on iOS or like with a basic `<TextInput />`.

**CodeSandBox/Snack link**
https://snack.expo.io/@vbriand/nativebase-paddingtop-input-bug

**Screenshots**
On top is the `<Input />` component and below a `<TextInput />` with a roughly similar styling.  

![Kapture 2021-06-30 at 22 54 43](https://user-images.githubusercontent.com/678530/124030368-67b48c80-d9f6-11eb-8934-82dd26e4243c.gif)

On iOS it works fine.  

![Kapture 2021-06-30 at 22 55 26](https://user-images.githubusercontent.com/678530/124030826-d98cd600-d9f6-11eb-98a0-4816deba4a4d.gif)

**Additional information**
- Only on Android
- Tested on 3.0.2 and 3.0.3
- `padding={0}` works though

**Comment by gretchelin** [comment]
I have the same problem. Setting any specific padding props (ie. py, px, pt, etc) on the component does not work, only general `padding` would.

My workaround is to use `extendTheme` and override `Input.baseStyle._android.p` to object value (empty object will do) and then specific padding props would finally work on the component.  (Sample code given below)

```
const theme = extendTheme({
  components: {
    Input: {
      baseStyle: {
        _android: {
          p: {}, // workaround to enable specific padding prop to work. value must be an object, but value inside this object is not being respected, so just pass empty object
          // now we can set px, py, pt, etc. here and it will be respected
        },
      },
    },
  },
});
```


This is how react-native-base set the style for android (code blow are taken from [NativeBase/src/theme/components/input.ts](https://github.com/GeekyAnts/NativeBase/blob/v3-pre-beta/src/theme/components/input.ts) at the time I write this)

```
const baseStyle = (props: Record<string, any>) => {
   ....//other code

  return {
     ....//other styles
    _android: {
          p: 3,
          _focus: {
            borderColor: 'primary.400',
          },
        },
        _ios: {
          px: 4,
          py: 3,
          _focus: {
            borderColor: 'primary.400',
          },
        },
    ...//other style
    };
};
```


I didn't take a look at how this lib actually extend the value so I don't know if this is the intended behavior or not. There is nothing in the documentation about this, though.

**Comment by intergalacticspacehighway** [comment]
Thanks for reporting and debugging @gretchelin. We've found an issue in the order of prop resolution.
The fix has been targeted for 3.1 version. We'll share detailed changelog for the same. 
As of now, we'll add a patch to fix the above issue for Input in 3.0.4.

**Comment by intergalacticspacehighway** [comment]
This has been fixed in the latest version with a workaround. We're changing the prop resolution for the next minor release. That would fix the issue at the core level.
